### PR TITLE
some doc formatting issues in passes/synthesis/unitary_synthesis.py

### PR DESCRIPTION
**Pull request by 1ucian0**
By accident I ran into some small formatting and spelling issues in `qiskit/transpiler/passes/synthesis/unitary_synthesis.py`.

### CODE REVIEW

File: qiskit/transpiler/passes/synthesis/unitary_synthesis.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
             Output dag with UnitaryGates synthesized to target basis.
 
         Raises:
-            TranspilerError: if a 'method' was specified for the class and is not
+            TranspilerError: if ``'method'`` was specified for the class and is not
```
Comment (jakelishman) [inline qiskit/transpiler/passes/synthesis/unitary_synthesis.py:246]: This one was either already fine, or the inner quotes in the changed version should be dropped.

Comment (1ucian0) [inline qiskit/transpiler/passes/synthesis/unitary_synthesis.py:246]:
```suggestion
            TranspilerError: if ``method`` was specified for the class and is not
```

Comment (1ucian0) [reply qiskit/transpiler/passes/synthesis/unitary_synthesis.py]: inner quotes removed in https://github.com/Qiskit/qiskit-terra/pull/9219/files/70bcae5d2d812014502c7b1aea6f6f4eee26e45a#r1038036824 thanks!

File: qiskit/transpiler/passes/synthesis/unitary_synthesis.py [hunk 2]
```diff
@@ -264,7 +264,7 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         if self.method == "default":
             # If the method is the default, we only need to evaluate one set of keyword arguments.
             # To simplify later logic, and avoid cases where static analysis might complain that we
-            # haven't initialised the "default" handler, we rebind the names so they point to the
+            # haven't initialised the "default" handler, we rebind the names, so they point to the
```
Comment (1ucian0) [inline qiskit/transpiler/passes/synthesis/unitary_synthesis.py:267]:
```suggestion
            # haven't initialised the "default" handler, we rebind the names so they point to the
```

[the remark this answers is not in the document]
Comment (1ucian0) [reply qiskit/transpiler/passes/synthesis/unitary_synthesis.py]: oh.. probably my IDE spellchecker, undone in https://github.com/Qiskit/qiskit-terra/pull/9219/commits/3bd338580f79ee900d214e181ee6957a4c4d2030

**Review by jakelishman — APPROVED** [review]
Looks good to me - it's good to squash these minor things.

**Pull request merged**